src/taipy/core/data/excel.py

```diff
@@ -237,56 +234,46 @@ def _read_as_numpy(self):
             return {sheet_name: df.to_numpy() for sheet_name, df in sheets.items()}
         return sheets.to_numpy()
 
-    def _read_as_pandas_dataframe(self, sheet_names=None) -> Union[Dict[Union[int, str], pd.DataFrame], pd.DataFrame]:
+    def _do_read_excel(self, engine, sheet_names, kwargs) -> pd.DataFrame:
+        df = pd.read_excel(
+            self._path,
+            sheet_name=sheet_names,
+            **kwargs,
+        )
+        if isinstance(df, dict):
```

Comment:
Why not make it check:
```suggestion
        if engine == "modin:
                if isinstance(df, dict):
                        for key, value in df.items():
                                ...
                else:
                        modin_pd.DataFrame(df)
```
So that if the engine is `pandas`, we can just return immediately?